src/main/java/com/minecolonies/coremod/entity/ai/citizen/shepherd/EntityAIWorkShepherd.java

```diff
@@ -0,0 +1,504 @@
+     */
+    protected static final int MAX_DISTANCE_FROM_HUT = 25;
+
+    /**
+     * Experience given per sheep sheared.
+     */
+    protected static final double EXP_PER_SHEEP = 5.0;
+
+    /**
+     * Distance two sheep need to be inside to breed.
+     */
+    protected static final int DISTANCE_TO_BREED = 10;
+
+    /**
+     * Amount of sheep to keep per Hut Level.
+     */
+    protected static final int MAX_SHEEP_MULTIPLIER = 2;
+
+    /**
+     * Sets up some important skeleton stuff for every ai.
+     *
+     * @param job the job class.
+     */
+    public EntityAIWorkShepherd(@NotNull final JobShepherd job)
+    {
+        super(job);
+        super.registerTargets(
+          new AITarget(IDLE, START_WORKING),
+          new AITarget(START_WORKING, this::startWorkingAtOwnBuilding),
+          new AITarget(PREPARING, this::prepareForShepherding),
+          new AITarget(SHEPHERD_SEARCH_FOR_SHEEP, this::searchForSheep),
+          new AITarget(SHEPHERD_DECIDE, this::decideWhatToDo),
+          new AITarget(SHEPHERD_SHEAR_SHEEP, this::shearSheep),
+          new AITarget(SHEPHERD_BREED_SHEEP, this::breedSheep),
+          new AITarget(SHEPHERD_BUTCHER_SHEEP, this::butcherSheep)
+        );
+        worker.setCanPickUpLoot(true);
+    }
+
+    /**
+     * Redirects the shepherd to his building.
+     *
+     * @return the next state.
+     */
+    private AIState startWorkingAtOwnBuilding()
+    {
+        worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.shepherd.goingToHut"));
+        if (walkToBuilding())
+        {
+            return getState();
+        }
+        return PREPARING;
+    }
+
+    /**
+     * Prepares the shepherd for fishing and
+     * requests shear
+     *
+     * @return the next AIState
+     */
+    private AIState prepareForShepherding()
+    {
+        if (checkForToolOrWeapon(ToolType.SHEARS) && checkForToolOrWeapon(ToolType.AXE) && checkOrRequestItems(new ItemStack(Items.WHEAT, 2)))
+        {
+            return getState();
+        }
+        if (job.getSheep().isEmpty())
+        {
+            return SHEPHERD_SEARCH_FOR_SHEEP;
+        }
+        return SHEPHERD_DECIDE;
+    }
+
+    /**
+     * If the job class has no sheep object the shepherd should search sheep.
+     *
+     * @return the next AIState the shepherd should switch to, after executing this method.
+     */
+    private AIState decideWhatToDo()
+    {
+        job.removeDeadSheep();
+
+        if (job.getSheep().isEmpty())
+        {
+            return SHEPHERD_SEARCH_FOR_SHEEP;
+        }
+
+        worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.shepherd.deciding"));
+
+        final EntitySheep shearingSheep = job.getSheep().stream().filter(sheepie -> !sheepie.getSheared()).findFirst().orElse(null);
+
+        int numOfBreedableSheep = job.getSheep().stream().filter(sheepie -> sheepie.getGrowingAge() == 0).toArray().length;
+
+        //setDelay(80);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)